internal/cloudapi/v2/imagerequest.go

```diff
@@ -69,10 +67,14 @@ func newAWSTarget(options UploadOptions, imageType distro.ImageType) (*target.Ta
 		ShareWithAccounts: awsUploadOptions.ShareWithAccounts,
 		BootMode:          amiBootMode,
 	})
+	// There's two possibilities for the ImageName: The user have provided a customization or not. In both cases we're
+	// going to need to append a UUID to make the name unique. Otherwise AWS will reject the image registration. The
+	// UUID is appended in the Register function so that the raw value of the name can be used for
+	// other purpose (Tags for instance).
 	if awsUploadOptions.SnapshotName != nil {
 		t.ImageName = *awsUploadOptions.SnapshotName
 	} else {
-		t.ImageName = key
+		t.ImageName = "composer-api"
```

Comment:
There's only two fields that I can use to pass down data to the `Register` function. `Key` and `ImageName`.

I've got a problem with making the ImageName unique:
If I append the uuid in this function, the Name tag and the AMI name, both have it, it breaks onPrem.
If I append the uuid in the `Register` function, it breaks onPrem.

I'll try to repurpose the `Key` field to use it as a mean to transmit a tagName. Since in the registration process the key is only used for the S3 upload which gets deleted after the image got registered.

Reply:
Repurposing was a bad idea, instead I'm adding a new optional `TagName` field in the `AWSTargetOptions`.
Then if it's provided, the `Register` function will use it, if not it'll default to the `ImageName`.
That should hopefully work.